modules/ckanext-sixodp_showcase/ckanext/sixodp_showcase/views.py

```diff
@@ -0,0 +1,307 @@
+                u'pkg': pkg,
+                u'resources_json': resources_json,
+                u'form_snippet': form_snippet,
+                u'errors_json': errors_json
+            }
+        )
+
+    def post(self, id):
+        context = self._prepare(id)
+        showcase_utils.check_edit_view_auth(id)
+
+        data_dict = dataset.clean_dict(
+            dataset.dict_fns.unflatten(
+                dataset.tuplize_dict(dataset.parse_params(tk.request.form))))
+        data_dict.update(
+            dataset.clean_dict(
+                dataset.dict_fns.unflatten(
+                    dataset.tuplize_dict(dataset.parse_params(
+                        tk.request.files)))))
+
+        data_dict['id'] = id
+        try:
+            pkg = tk.get_action('ckanext_showcase_update')(context, data_dict)
+        except tk.ValidationError as e:
+            errors = e.error_dict
+            error_summary = e.error_summary
+            return self.get(id, data_dict, errors, error_summary)
+
+        tk.c.pkg_dict = pkg
+
+        # redirect to showcase details page
+        url = h.url_for('sixodp_showcase.read', id=pkg['name'])
+        return h.redirect_to(url)
+
+
+def manage_datasets(id):
+    return showcase_utils.manage_datasets_view(id)
+
+
+def delete(id):
+    return showcase_utils.delete_view(id)
+
+
+def dataset_showcase_list(id):
+    return showcase_utils.dataset_showcase_list(id)
+
+
+def admins():
+    return showcase_utils.manage_showcase_admins()
+
+
+def admin_remove():
+    return showcase_utils.remove_showcase_admin()
+
+
+def upload():
+    return showcase_utils.upload()
+
+
+showcase = Blueprint(u'sixodp_showcase', __name__)
+showcase.add_url_rule('/showcase', view_func=search)
+showcase.add_url_rule('/showcase/new', view_func=CreateView.as_view('new'))
+showcase.add_url_rule('/showcase/edit/<id>',
+                      view_func=EditView.as_view('edit'),
+                      methods=[u'GET', u'POST'])
+
+showcase.add_url_rule('/showcase/delete/<id>',
+                      view_func=delete,
+                      methods=[u'GET', u'POST'])
+showcase.add_url_rule('/showcase/manage_datasets/<id>',
+                      view_func=manage_datasets,
+                      methods=[u'GET', u'POST'])
+showcase.add_url_rule('/dataset/showcases/<id>',
+                      view_func=dataset_showcase_list,
+                      methods=[u'GET', u'POST'])
+showcase.add_url_rule('/ckan-admin/showcase_admins',
+                      view_func=admins,
+                      methods=[u'GET', u'POST'])
+showcase.add_url_rule('/ckan-admin/showcase_admin_remove',
+                      view_func=admin_remove,
+                      methods=[u'GET', u'POST'])
+showcase.add_url_rule('/showcase_upload',
+                      view_func=upload,
+                      methods=[u'POST'])
+showcase.add_url_rule('/showcase/<id>', view_func=read)
+
+
+def get_blueprints():
+    return [showcase] + views.get_blueprints()
```

Comment:
I added original blueprints from `ckanext-showcase` in here because they were not registered in application and they are needed by some original methods (like manage_datasets etc). I'm still confused why routes (`showcase_blueprint`) from original extension aren't registered, because I'm pretty sure those were working before (could it be something on my vagrant environment, or did I just miss these before) 🤔